Mathlib/Algebra/Category/Grp/Ulift.lean

```diff
@@ -0,0 +1,137 @@
+/-
+Copyright (c) 2024 Sophie Morel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sophie Morel
+-/
+import Mathlib.Algebra.Category.Grp.Limits
+import Mathlib.CategoryTheory.Limits.Preserves.Ulift
+import Mathlib.CategoryTheory.Preadditive.AdditiveFunctor
+
+/-!
+This file shows that the functors `Grp.uliftFunctor` and `CommGrp.uliftFunctor`
+(as well as the additive versions) are fully faithful, preserves all limits and
+create small limits.
+
+It also shows that `AddCommGrp.uliftFunctor` preserves zero morphisms and is an additive functor.
+
+-/
+
+universe v w w' u
+
+open CategoryTheory Limits
+
+namespace Grp
+
+/-- The universe lift functor for groups is faithful.
+-/
+@[to_additive
+  "The universe lift functor for additive groups is faithful."]
+instance : uliftFunctor.{u, v}.Faithful where
+  map_injective {_ _ f g} heq  := by
+    ext a
+    apply_fun (fun h ↦ h {down := a}) at heq
+    change Equiv.ulift.symm (f a) = Equiv.ulift.symm (g a) at heq
+    exact (EmbeddingLike.apply_eq_iff_eq _).mp heq
+
+/-- The universe lift functor for groups is full.
+-/
+@[to_additive
+  "The universe lift functor for additive groups is full."]
+instance : uliftFunctor.{u, v}.Full where
+  map_surjective f :=
+    ⟨ofHom (MonoidHom.mk (OneHom.mk (fun x => (f (ULift.up x)).down)
+    (by change (f 1).down = ?_; rw [f.map_one]; rfl))
+    (fun _ _ ↦ by simp only [uliftFunctor_obj, coe_of];
+                  change (f (_ * _)).down = _; rw [f.map_mul]; rfl)), rfl⟩
+
+@[to_additive]
+noncomputable instance uliftFunctor_preservesLimit {J : Type w} [Category.{w'} J]
+    (K : J ⥤ Grp.{u}) : PreservesLimit K uliftFunctor.{v, u} where
+  preserves lc := ReflectsLimit.reflects (F := forget Grp.{max u v}) (IsLimit.ofIsoLimit
+    (Classical.choice (PreservesLimit.preserves (F := CategoryTheory.uliftFunctor) (Classical.choice
+    (PreservesLimit.preserves (F := forget Grp) lc)))) (Cones.ext (Iso.refl _) (fun _ ↦ rfl)))
```

Comment:
The correct way to use the reflects/preserves limits API is via the definitions `isLimitOfReflects/Preserves`:
```suggestion
  preserves lc := ⟨isLimitOfReflects (forget Grp.{max u v})
    (isLimitOfPreserves CategoryTheory.uliftFunctor (isLimitOfPreserves (forget Grp) lc))⟩
```